### PR TITLE
upgpkg(main/nodejs): update to v18.2.0

### PR DESCRIPTION
**Pull request by thunder-coding**
Do not expect to get this merged soon. Will need to test this and maybe also fix builds

### CODE REVIEW

**Comment by thunder-coding** [comment]
v18.0.0 seems quite production ready! There doesn't seem to be any new regressions or bugs (except for the getaddrinfo one for which a workaround has been applied. Note that this was not a Node.js bug but a bug in Bionic on some devices, needs more confirmation on the list of devices affected). Will catch up with v18.2.0 in the same PR

File: packages/nodejs/lib-os.js.patch [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+ const {
++  getNumCPUs,
+   getCPUs,
+   getFreeMem,
+   getHomeDirectory: _getHomeDirectory,
+@@ -114,6 +115,10 @@
+   return [avgValues[0], avgValues[1], avgValues[2]];
+ }
+ 
++function numcpus() {
```
Comment (thunder-coding) [inline packages/nodejs/lib-os.js.patch:15]: This is a ugly fix for concurrency deadlock when passing a directory to `node --test`. See `src-node_os.cc.patch` and `lib-os.js.patch` for more information. This is a **non-standard** function, relying on this is bad. libuv 1.44 seems to have a `uv_available_parallelism` which Node.js is willing to adopt for getting concurrency limits as seen in lib/internal/test_runner/test.js:34

File: packages/nodejs/src-getaddrinfo.patch [hunk 2]
```diff
@@ -0,0 +1,48 @@
+--- /dev/null	1972-07-31 19:09:33.659999999 +0530
```
Comment (thunder-coding) [inline packages/nodejs/src-getaddrinfo.patch:1]: This is fix for getaddrinfo on some devices where querying for localhost with IPv6 fails with EA_NODATA. A bug in netd somewhere

**Comment by thunder-coding** [comment]
Overall, the test suit is quite green with 30 failing tests, 2 recieving OOM when run parallel (when executed alone, they pass), and 1 crashing due to usage of `setuid`. Also much of the failing tests actually have either hard-coded paths or treat Android as non-Linux OS (aka Windows in most cases)

**Comment by thunder-coding** [comment]
I would like to request reviews form @Grimler91 and @fornwall regarding the two ugly fixes in this PR as they have earlier maintained Node.js.

**Review by Grimler91 — APPROVED** [review]
Definitely not the ugliest hacks we have, haven't tested it, but code looks good to me!

**Pull request merged**